src/blueprint/from_config.rs

```diff
@@ -52,21 +53,92 @@ fn to_directive(const_directive: ConstDirective) -> Valid<Directive> {
+
+  // Handle Base URL setup
+  server = handle_base_url(server, config)?;
+
+  // Configure other server settings
+  server = configure_server(server, config);
+
+  Valid::Ok(server.clone())
+}
+
+fn handle_graphiql(mut server: Server, config: &Config) -> Valid<Server> {
+  if let Some(enable_graphiql) = config.server.enable_graphiql.clone() {
     let lowered_route = enable_graphiql.to_lowercase();
     if RESTRICTED_ROUTES.contains(&lowered_route.as_str()) {
-      return Valid::fail(format!(
-        "Cannot use restricted routes '{}' for enabling graphiql",
-        enable_graphiql
-      ))
-      .trace("enableGraphiql")
-      .trace("@server")
-      .trace("schema");
+      return Err(
+        ValidationError::new(format!(
+          "Cannot use restricted routes '{}' for enabling graphiql",
+          enable_graphiql
+        ))
+        .trace("enableGraphiql")
+        .trace("@server")
+        .trace("schema"),
+      );
+    } else {
+      server = server.clone().enable_graphiql(Some(enable_graphiql));
     }
   }
+  Ok(server)
+}
+
+fn handle_response_headers(mut server: Server, config: &Config) -> Valid<Server> {
+  let headers = config
+    .server
+    .response_headers
+    .0
+    .clone()
+    .validate_all(|(k, v)| {
+      let name = HeaderName::from_bytes(k.as_bytes())
+        .map_err(|e| ValidationError::new(format!("Parsing failed because of {}", e)));
+      let value =
+        HeaderValue::from_str(v.as_str()).map_err(|e| ValidationError::new(format!("Parsing failed because of {}", e)));
+      name.validate_both(value)
+    })
+    .trace("responseHeaders")
+    .trace("@server")
+    .trace("schema")?;
+
+  let mut response_headers = HeaderMap::new();
+  response_headers.extend(headers);
+  server = server.clone().response_headers(response_headers);
+  Ok(server)
+}
+
+fn handle_base_url(mut server: Server, config: &Config) -> Valid<Server> {
+  if let Some(base_url) = config.server.base_url.clone() {
+    let _ = Valid::Ok(reqwest::Url::parse(base_url.as_str()).map_err(|e| ValidationError::new(e.to_string()))?);
+    server = server.clone().base_url(Some(base_url));
+  }
+  Ok(server)
+}
+
+fn configure_server(mut server: Server, config: &Config) -> Server {
+  server = server
+    .clone()
+    .allowed_headers(config.server.allowed_headers.clone())
+    .batch(config.server.batch.clone())
+    .enable_apollo_tracing(config.server.enable_apollo_tracing)
+    .enable_cache_control_header(config.server.enable_cache_control_header)
+    .enable_http_cache(config.server.enable_http_cache)
+    .enable_introspection(config.server.enable_introspection)
+    .enable_query_validation(config.server.enable_query_validation)
+    .enable_response_validation(config.server.enable_response_validation)
+    .global_response_timeout(config.server.global_response_timeout)
+    .port(config.server.port)
+    .upstream(config.server.upstream.clone())
+    .vars(config.server.vars.clone().0);
 
-  Valid::Ok(())
+  server
 }
```

Comment:
Can `impl From<config::Server> for blueprint::Server

Reply:
This should be tryFrom and return Valid<Self>